File: backend/app/schemas/vehicle.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class VehicleBase(BaseModel):
    plate: str
    brand: Optional[str] = None
    model: Optional[str] = None
    kilometers: Optional[int] = None
    color: Optional[str] = None
    imported: Optional[bool] = None
    description: Optional[str] = None
    engineSize: Optional[str] = None
    fuelType: Optional[str] = None

class VehicleCreate(VehicleBase):
    customer_id: int

class VehicleUpdate(BaseModel):
    plate: Optional[str] = None
    brand: Optional[str] = None
    model: Optional[str] = None
    kilometers: Optional[int] = None
    color: Optional[str] = None
    imported: Optional[bool] = None
    description: Optional[str] = None
    engineSize: Optional[str] = None
    fuelType: Optional[str] = None

class Vehicle(VehicleBase):
    id: int
    customer_id: int
    deleted_at: Optional[datetime] = None  

class VehicleWithCustomer(VehicleBase):
    id: int
    customer_id: int
    customer_name: Optional[str] = None
    deleted_at: Optional[datetime] = None

    class Config:
        from_attributes = True

File: frontend/management_app/src/interfaces/Vehicle.ts
// Interface para veículo
export interface Vehicle {
  id: number;
  plate: string;
  brand: string;
  model: string;
  kilometers: number;
  customer_id: number;
  description?: string;
  color?: string;
  imported?: boolean;
  engineSize?: string;
  fuelType?: string;
  created_at: string;
  updated_at: string;
  deleted_at?: string;
}

// Interface para criar veículo
export interface VehicleCreate {
  plate: string;
  brand: string;
  model: string;
  kilometers: number;
  customer_id: number;
  description?: string;
  color?: string;
  imported?: boolean;
  engineSize?: string;
  fuelType?: string;
}

// Interface para atualizar veículo
export interface VehicleUpdate {
  plate?: string;
  brand?: string;
  model?: string;
  kilometers?: number;
  customer_id?: number;
  description?: string;
  color?: string;
  imported?: boolean;
  engineSize?: string;
  fuelType?: string;
}

// Interface para mapa de contagem de veículos
export interface VehicleCountMap {
  [customerId: string]: number;
}

// Interface para veículo com nome do cliente
export interface VehicleWithCustomer extends Vehicle {
  customer_name?: string;
}